sklearn/datasets/_samples_generator.py

```diff
@@ -672,6 +672,15 @@ def make_regression(
         return X, y
 
 
+@validate_params(
+    {
+        "n_samples": [Interval(Integral, 0, None, closed="left")],
+        "shuffle": ["boolean"],
+        "noise": [Interval(Real, 0, None, closed="left"), None],
+        "random_state": ["random_state"],
+        "factor": [Interval(Real, 0, 1, closed="both")],
```

Comment:
According to the validation on line 729, 0 is allowed and 1 is not.
```suggestion
        "factor": [Interval(Real, 0, 1, closed="left")],
```
You can now remove validation on line 729, and check in test_sample_generator if we check for this specific error (in which case you can remove it). 

Please also fix the param description in the docstring which should be `ìn the range [0, 1)`